src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,36 @@
 package mate.academy;
 
+import java.util.Map;
+import java.util.Random;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> getUserData(userId), executor);
+    }
+
+    private UserData getUserData(String userId) {
+        try {
+            TimeUnit.MILLISECONDS.sleep(new Random(4000).nextLong());
```

Comment:
Using a new Random instance with a fixed seed will yield the same long value on each call, which is probably not the intended behavior for simulating a delay. Consider using a Random instance without a fixed seed or use `ThreadLocalRandom.current().nextLong(upperBound)` to generate a random delay up to an upper bound.